tools/spice/spice_utils.py

```diff
@@ -0,0 +1,159 @@
+import logging
+import os
+import re
+from datetime import datetime, timedelta
+from pathlib import Path
+from typing import Optional
+
+import spiceypy as spice
+from spiceypy.utils.exceptions import NotFoundError
+
+logging.basicConfig(level=logging.ERROR)
+logger = logging.getLogger(__name__)
+
+
+class SpiceKernelManager:
+    """Manages the loading and handling of SPICE kernels.
+
+    Parameters
+    ----------
+    kernel_path : Path
+        The local directory path where the SPICE kernel files are stored.
+    extensions : list of str, optional
+        A list of file extensions to be considered as valid SPICE kernel files.
+    """
+
+    def __init__(self, kernel_path: Path, extensions: Optional[list[str]] = None):
+        self.kernel_path = kernel_path
+        if extensions is None:
+            self.extensions = [".bpc", ".bsp", ".ti", ".tf", ".tls", ".tsc"]
+        else:
+            self.extensions = extensions
+
+    def furnsh(self):
+        """Load a kernel file into the SPICE system."""
+        self.clear()
+        files_loaded = []
+
+        for filename in os.listdir(self.kernel_path):
+            if filename.endswith(tuple(self.extensions)):
+                filepath = os.path.join(self.kernel_path, filename)
+                spice.furnsh(filepath)
+                files_loaded.append(filename)
+            else:
+                logger.info(f"Invalid kernel_type extension: {filename}.")
+
+        return files_loaded
+
+    def clear(self):
+        """Clear the SPICE kernel pool."""
+        logger.info("Clearing SPICE kernel pool.")
+        spice.kclear()
+
+
+def ls_kernels(extensions=None) -> list:
```

Comment:
Be explicit with the names.
```suggestion
def list_loaded_kernels(extensions=None) -> list:
```